db/migrate/20200611090846_add_task_lock_index_on_resource_type_and_task_id.rb

```diff
@@ -0,0 +1,5 @@
+class AddTaskLockIndexOnResourceTypeAndTaskId < ActiveRecord::Migration[6.0]
+  def change
+    add_index :foreman_tasks_locks, [:resource_type, :task_id]
```

Comment:
Should we add `:resource_id` as well?

Reply:
updated, and cleaned up a couple of indexes that are no longer required as they are covered by other indexes (as postgres can use leading columns in an index for searching as well)